src/textual/app.py

```diff
@@ -1735,13 +1734,16 @@ def _register(
         if not widgets:
             return []
 
-        new_widgets = list(widgets)
+        widget_list = list(widgets)
```

Comment:
This isn't great. But the original was problematic.

It is doing a completely superfluous conversion from tuple to list. Suggest that we don't need `widget_list`, and that `new_widgets` should be either `reversed(widgets)` or just `widgets`. And type as an iterable.

Reply:
You are absolutely right! That is the absolute most obvious thing to do and I was sure I had tried typing the result of `reversed` as an iterable and it had complained...
But I just did it correctly, so I must have tried the wrong thing!